### PR TITLE
DOC: Add examples for creating an index accessor

### LINKED ISSUE #49202: DOC: Improve the register_index_accessor Documentation
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.api.extensions.register_index_accessor.html#pandas.api.extensions.register_index_accessor

### Documentation problem

There is no example showing how to create or what is required on the class to make an index accessor.

### Suggested fix for documentation

Add a link to an example or expand the topic to include a how to implement this.  Also document what is required to make a custom indexer.

### PR DESCRIPTION
**Pull request by sjalkote**
Picked up issue #49202. This adds examples for creating index, dataframe, and series accessors.
- [X] closes #49202 
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by sjalkote** [comment]
pre-commit.ci autofix

**Comment by sjalkote** [comment]
pre-commit.ci autofix

File: pandas/core/accessor.py [hunk 1]
```diff
@@ -318,20 +293,138 @@ def decorator(accessor):
...
+    class IntAccessor:
+        def __init__(self, pandas_obj):
+            if not all(pandas_obj[col].dtype == 'int64' for col in pandas_obj.columns):
+                raise AttributeError("All columns must contain integer values only")
+            self._obj = pandas_obj
+
+        def sum(self):
+            return self._obj.sum()
+
+
```
Comment (s1099) [inline pandas/core/accessor.py:316]:
```suggestion

```

File: pandas/core/accessor.py [hunk 2]
```diff
@@ -318,20 +293,138 @@ def decorator(accessor):
...
```
Comment (s1099) [inline pandas/core/accessor.py:362]:
```suggestion

```

**Review by s1099 — CHANGES_REQUESTED** [review]
remove double newline

**Review by datapythonista — COMMENTED** [review]
`Appender` is something that we used before `doc` existed, but I think it'd be better to not use, as makes things more difficult to understand. I think you're adding the examples to all register accessor functions, so you can add the `Examples` header to the main docstring, and then add a variable for the content of the section as it's done with `{klass}`.

Unrelated, in the examples section is not needed to create a `code-block`. You can just start the code with `>>>` and the docstring parser should understand it's code. Also, it's not needed to import pandas, `pd` should already be available without importing.

**Comment by sjalkote** [comment]
pre-commit.ci autofix

**Comment by datapythonista** [comment]
Can you have a look at the CI errors please?

File: pandas/core/accessor.py [hunk 3]
```diff
@@ -318,21 +293,116 @@ def decorator(accessor):
...
+        'row2': {'1':2, '2':'b'}
+        },orient='index')
+>>> df.index.int_accessor
+Traceback (most recent call last):
+...
+AttributeError: The index must only be an integer value.
+>>> df = pd.DataFrame({
+        'col1': [1, 2, 3, 4],
+        'col2': ['a', 'b', 'c', 'd']
+        }, index=[1, 2, 5, 8])
```
Comment (s1099) [inline pandas/core/accessor.py:399]:
```suggestion
>>> df = pd.DataFrame(
...     {"col1": [1, 2, 3, 4], "col2": ["a", "b", "c", "d"]}, index=[1, 2, 5, 8]
... )
```

File: pandas/core/accessor.py [hunk 4]
```diff
@@ -318,21 +293,116 @@ def decorator(accessor):
...
+
+        def even(self):
+            return [x for x in self._obj if x % 2 == 0]
+
+Back in an interactive IPython session:
+
+>>> df = pd.DataFrame.from_dict({
+        'row1': {'1':1, '2':'a'},
+        'row2': {'1':2, '2':'b'}
+        },orient='index')
```
Comment (s1099) [inline pandas/core/accessor.py:391]:
```suggestion
>>> df = pd.DataFrame.from_dict(
...     {"row1": {"1": 1, "2": "a"}, "row2": {"1": 2, "2": "b"}}, orient="index"
... )
```

**Review by s1099 — CHANGES_REQUESTED** [review]

**Comment by Aloqeely** [comment]
The checks are still failing, could you please fix them?

**Comment by sjalkote** [comment]
Hi @datapythonista I believe we have fixed all the CI errors. Could you please re-review?

File: ci/code_checks.sh [hunk 5]
```diff
@@ -598,6 +598,9 @@ if [[ -z "$CHECK" || "$CHECK" == "docstrings" ]]; then
         -i "pandas.api.extensions.ExtensionArray.unique RT03,SA01" \
         -i "pandas.api.extensions.ExtensionArray.view SA01" \
         -i "pandas.api.extensions.register_extension_dtype SA01" \
+        -i "pandas.api.extensions.register_index_accessor" \
```
[the remark this answers is not in the document]
Comment (sjalkote) [reply ci/code_checks.sh]: Yep, I removed it. The builds are still passing.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @sjalkote

**Pull request merged**